### PR TITLE
Crypto iteration 1

### PR DESCRIPTION
**Pull request by arthurchan35**
hey Brian,

following our discussion here is the first iteration pull. 
We have also kept `CryptoHashGenerationResponse` and `CryptoHashVerificationContext` since we felt those are necessary and they are just a "record class" to aggregate the information in a more convenient way. 

Bests. 
Carlos.

### CODE REVIEW

File: modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/CryptoHasherImpl.java [hunk 1]
```diff
@@ -0,0 +1,186 @@
+/**
```
Comment (liferay-appsec) [inline modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/CryptoHasherImpl.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

**Review by liferay-appsec — COMMENTED** [review]
This is an automatic review suggestion

**Comment by arthurchan35** [comment]
ci:test:relevant

**Comment by arthurchan35** [comment]
ci:test:sf

**Comment by arthurchan35** [comment]
ci:test:relevant

**Comment by arthurchan35** [comment]
ci:test:sf

**Comment by csierra** [comment]
ci:forward

**Pull request closed without merging**